mmcv/ops/feature_refine.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+from torch.autograd import Function
+from torch.autograd.function import once_differentiable
+
+from ..utils import ext_loader
+
+ext_module = ext_loader.load_ext(
+    '_ext', ['feature_refine_forward', 'feature_refine_backward'])
+
+
+class FeatureRefineFunction(Function):
+    """Feature refine class."""
+
+    @staticmethod
+    def forward(ctx, features, best_rbboxes, spatial_scale, points=1):
+        """Forward function."""
+        ctx.spatial_scale = spatial_scale
+        ctx.points = points
+        ctx.save_for_backward(best_rbboxes)
+        assert points in [1, 5]
+        assert features.is_cuda
```

Comment:
Better not is_cuda assert here since we might have cpu or other device implementation in future.